core/blockchain.go

```diff
@@ -1999,12 +2052,146 @@ func (bc *BlockChain) reorg(oldBlock, newBlock *types.Block) error {
 	return nil
 }
 
-// futureBlocksLoop processes the 'future block' queue.
-func (bc *BlockChain) futureBlocksLoop() {
-	defer bc.wg.Done()
+// InsertBlock executes the block, runs the necessary verification
+// upon it and then persist the block and the associate state into the database.
+// The key difference between the InsertChain is it won't do the canonical chain
+// updating. It relies on the additional SetChainHead call to finalize the entire
+// procedure.
+func (bc *BlockChain) InsertBlock(block *types.Block) error {
+	if !bc.chainmu.TryLock() {
+		return errChainStopped
+	}
+	defer bc.chainmu.Unlock()
 
+	return bc.insertBlock(block)
+}
```

Comment:
IMO this is not very great: we already have `InsertChain`, so instead of duplicating that logic, we should simply use it, and just send a 1-element slice if we only have one block to insert. 
There are a few differences, e.g. this method will not postpone-for-later future blocks, but instead return errors for them. Not sure what the intent behind that change is, nor why this was split out in general.